app/models/import_prerequisite.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class ImportPrerequisite < ApplicationRecord
+  # A digital_object_import actually is required on save, but it can be supplied as a
+  # digital_object_import_id rather than a digital_object_import.
+  belongs_to :digital_object_import, required: false
+  belongs_to :batch_import
+  belongs_to :prerequisite_digital_object_import, class_name: 'DigitalObjectImport'
+
+  # Note, below, that we're requiring digital_object_import_id and prerequisite_digital_object_import_id
```

Comment:
I originally thought so too, but it turns out that the model wouldn't save.  It you do a `presence` check for `digital_object_import`, `presence` apparently validates the presence of the `digital_object_import` rather than `digital_object_import_id`.

The reason this was a problem is that in `batch_import_start_job.rb`, I have a potentially large number of `digital_object_import_id` and `prerequisite_digital_object_import_id` numeric values and I didn't want to do a ton of ActiveRecord lookups for all of those objects.  That's why I needed to use IDs.  One alternative would have been to do a bunch of literal SQL insert queries instead, but I wasn't hesitant to totally bypass all ActiveRecord validations.  I guess it might not be the worst thing, since it's a fairly straightforward insert operation on guaranteed-existing records.  Do you think that would be worth it? 

One alternative that I considered was

Reply:
I definitely prefer your solution to bypassing ActiveRecord validation.

I just added a presence check on digital_object_import for batch_import and create a new digital object import by setting batch_import_id and it worked. I didn't get any errors. So, I am wondering if ActiveRecord is getting confused somehow. 

Maybe the `inverse_of` should be set, like described here: https://guides.rubyonrails.org/active_record_validations.html#presence

It also seems that since `require: true` is the default for belongs_to relationships the additional presence check isn't necessary because `require: true` automatically requires a presence check. 
https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-belongs_to

Reply:
Aha!  I was actually able to just change back to normal associations and it works.  I'm not sure what was going on before, but the code should now look more like you'd expect.  Tests are passing, so it looks like things are good.